testing/database.rst

```diff
@@ -112,26 +201,55 @@ the employee which gets returned by the ``Repository``, which itself gets
 returned by the ``EntityManager``. This way, no real class is involved in
 testing.
 
-Changing Database Settings for Functional Tests
------------------------------------------------
+Mocking a Doctrine Repository in Functional Tests
+-------------------------------------------------
 
-If you have functional tests, you want them to interact with a real database.
-Most of the time you want to use a dedicated database connection to make sure
-not to overwrite data you entered when developing the application and also
-to be able to clear the database before every test.
+In :ref:`functional tests <functional-tests>` you'll make queries to the
+database using the actual Doctrine repositories, instead of mocking them. To do
+so, get the entity manager via the service container as follows::
 
-To do this, you can override the value of the ``DATABASE_URL`` env var in the
-``phpunit.xml.dist`` to use a different database for your tests:
+    // tests/Repository/ProductRepositoryTest.php
+    namespace App\Tests\Repository;
 
-.. code-block:: xml
+    use App\Entity\Product;
+    use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
-    <?xml version="1.0" charset="utf-8" ?>
-    <phpunit>
-        <php>
-            <!-- the value is the Doctrine connection string in DSN format -->
-            <env name="DATABASE_URL" value="mysql://USERNAME:PASSWORD@127.0.0.1/DB_NAME"/>
-        </php>
-        <!-- ... -->
-    </phpunit>
+    class ProductRepositoryTest extends KernelTestCase
+    {
+        /**
+         * @var \Doctrine\ORM\EntityManager
+         */
+        private $entityManager;
+
+        protected function setUp()
+        {
+            $kernel = self::bootKernel();
+
+            $this->entityManager = $kernel->getContainer()
+                ->get('doctrine')
+                ->getManager();
+        }
+
+        public function testSearchByCategoryName()
```

Comment:
why not ByProductName ? as the fixture uses product->setName

also maybe a name without the `!` :)

Reply:
Yes, Maker bundle uses "AppFixtures" (https://github.com/symfony/maker-bundle/blob/020c04826455abf72184d487a1da65a2632f5fd3/src/Maker/MakeFixtures.php)